docs/source-2.0/aws/aws-core.rst

```diff
@@ -1179,6 +1179,290 @@ trait. Setting only the ``requestChecksumRequired`` property of the
+            @length(max: 128)
+            tags: TagList
+        }
+    }
+
+The following example shows a typical service with the default tagging
+operations. It can be understood that Forecast resource tags are managed
+through the operates attached to the service.
+
+.. code-block:: smithy
+
+    @tagEnabled
+    service Weather {
+        resources: [Forecast]
+        operations: [TagResource, UntagResource, ListTagsForResource]
+    }
+
+    @arn(template: "city/{cityId}/forecast/{forecastId}")
+    @taggable
+    resource Forecast {
+        identifiers: {
+            forecastId: ForecastId
+        }
+    }
+
+
+.. smithy-trait:: aws.api#taggable
+
+--------------------------
+``aws.api#taggable`` trait
+--------------------------
+
+Trait summary
+    Indicates the resource supports AWS tag associations and identifies resource
+    specific operations that perform CRUD on the associated tags. Managing tag
+    associations on a resource through service-wide operations is possible if
+    the resource has an :ref:`aws.api#arn-trait`.
+
+Trait selector
+    ``resource``
+Trait value
+    ``structure``
+
+The ``aws.api#taggable`` trait is a structure that supports the following
+members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 20 70
+
+    * - Property
+      - Type
+      - Description
+    * - property
+      - ``string``
+      - The name of the resource property representing tags for the resource.
+        Specifying this enables Smithy to understand which resource lifecycle
+        operations operate on tags.
+    * - apiConfig
+      - ``structure``
+      - Configuration structure for resource specific tagging operations, if
+        applicable.
+    * - apiConfig.tagApi
+      - ``ShapeID``
+      - Defines the operation used to create and update tags associations for
+        the resource. The value MUST be a valid :ref:`shape-id` that targets an
+        ``operation`` shape.
+    * - apiConfig.untagApi
+      - ``ShapeID``
+      - Defines the operation used to deletes tag associations from the
+        resource. The value MUST be a valid :ref:`shape-id` that targets an
+        ``operation`` shape.
+    * - apiConfig.listTagsApi
+      - ``ShapeID``
+      - Defines the operation used to list tags for the resource. The value
+        MUST be a valid :ref:`shape-id` that targets an ``operation`` shape.
+
+Resource specific tagging operations tagApi, untagApi, and listTagsApi have
+corresponding requirements to :ref:`tag-resource-api`, :ref:`untag-resource-api`,
+and :ref:`listtags-resource-api`. The differences is that these operations may
+have any name, and the resource ARN input member requirement is replaced by the
+expected identifier binding rules for instance operations.
+
+The following is an example of a resource with its own resource specific
+tagging operations. Note the service has disabled default tagging operations
+and the resource lacks an :ref:`aws.api#arn-trait`.
+
+.. code-block:: smithy
```

Comment:
this is great. Can we include an example earlier too that shows how services can use service-wide tagging operations + ARNs?

Reply:
The snippet right above this trait is that example. I didn't repeat the examples of TagResource, UntagResource, ListTagsForResource and the types associated it. After those sections the last example snippet just clearly shows the operations on the service, and the resource having the `taggable` trait along with the `arn` trait.